ibridges/path.py

```diff
@@ -380,6 +380,51 @@ def dataobject(self) -> irods.data_object.iRODSDataObject:
 
         raise irods.exception.DataObjectDoesNotExist(str(IrodsPath))
 
+    def open(self, mode="r", **kwargs):
+        """Open a data object for reading or writing.
+
+        Parameters
+        ----------
+        mode, optional
+            Whether to read or write, by default "r" meaning read. To write to
+            a data object, use "w", appending to a data object can be done with "a".*
+            Note that opening data objects is always done in binary
+            mode, so to write a string you need to encode it, while reading a string from
+            a data object requires you to decode it. You are advised to use a consistent
+            (utf-8) encoding for all your data objects.
+        kwargs:
+            Extra keyword arguments for the python-irodsclient to parse.
+
+        Returns
+        -------
+            A file handle to be used for reading or writing.
+
+        Raises
+        ------
+        ValueError
+            When the IrodsPath points to a collection.
+        DataObjectDoesNotExist:
+            When the data object does not exist and the read mode is given.
+
+        Examples
+        --------
+        >>> ipath = Irodspath(session, "some_obj.txt")
+        >>> with ipath.open("w") as handle:
+        >>>     handle.write("This is a test string".encode("utf-8"))
+        >>> with ipath.open("r") as handle:
+        >>>     print(handle.read().decode("utf-8))
+        This is a test string
+        >>> with ipath.open("a") as handle:
+        >>>     handle.write("A string appended at the end.".encode("utf-8"))
+
+        """
+        if self.collection_exists():
+            raise ValueError("Cannot open collection, only data objects can be opened.")
+        # Create the data object if it does not exist.
+        if mode == "w" and not self.dataobject_exists():
```

Comment:
Can we add that existing object cannot be opened in write mode. It would be very useful as it led to inconsistencies when I tried that previously!!!

Can we maybe add an else which throws a message that an existing object can only be opened in read mode?

Reply:
Or maybe we need to test a bit more for the writing existing data objects.

Reply:
That seems inconsistent with how opening files works though. Doesn't it make more sense that the data object gets removed and then opened for writing?

Reply:
@chStaiger I have tested, and writing to a data object that already exists is possible and will simply overwrite it. I think this is how it should be done. Let me know if you agree, then you can perhaps approve and merge.

Reply:
Yes, I think I opened the data object in "append" mode. And that led to problems. So all fine with me.